### PR TITLE
Added the azure ec2_run policy

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [ ] dependencies

##### In Azure, an instance is called a VM
1. Adding the azure VM run
2. Added the unit test cases for the vm_run policy
3. Added functionalities
     1. Created the main_operations: All policy runner calls will be taken from this file.
     2. Created a separate class for Azure/ was the main policy runners
4. The policy implementation UM diagram is attached below.
5. Any policy implementation needs to follow this approach. 

Updated UML dia
Policy Implementation-Policy Implementation (1)

Grafana Panel with combined of instance_run and ec2_run
Screenshot 2023-12-28 at 11 38 17 AM

### CODE REVIEW

File: cloud_governance/common/helpers/azure/azure_cleanup_operations.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+
```
Comment (ebattat) [inline cloud_governance/common/helpers/azure/azure_cleanup_operations.py:1]: Do you plan to change class name to AzurePolicyOperations ?

File: cloud_governance/common/helpers/cleanup_operations.py [hunk 2]
```diff
@@ -14,13 +14,27 @@ class AbstractCleanUpOperations(ABC):
```
Comment (ebattat) [inline cloud_governance/common/helpers/cleanup_operations.py:14]: Do you plan to change class name to PolicyOperations ?

File: cloud_governance/common/helpers/aws/aws_cleanup_operations.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import datetime
```
Comment (ebattat) [inline cloud_governance/common/helpers/aws/aws_cleanup_operations.py:1]: Do you plan to change class name to AWSPolicyOperations ?

File: cloud_governance/common/helpers/azure/azure_cleanup_operations.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        """
+        This method deletes the
+        :param resource_id:
+        :type resource_id:
+        :return:
+        :rtype:
+        """
+        action = "deleted"
+        try:
+            if self._policy == 'vm_run':
```
Comment (ebattat) [inline cloud_governance/common/helpers/azure/azure_cleanup_operations.py:43]: Are you still using vm_run instread of instance_run ?

File: docs/source/index.md [hunk 5]
```diff
@@ -11,7 +11,7 @@ This tool support the following policies:
 
 * Real time Openshift Cluster cost, User cost
 * [ec2_idle](../../cloud_governance/policy/aws/ec2_idle.py): idle ec2 in last 4 days, cpu < 2% & network < 5mb.
-* [ec2_run](../../cloud_governance/policy/aws/cleanup/ec2_run.py): running ec2.
+* [ec2_run](../../cloud_governance/policy/aws/cleanup/instance_run.py): running ec2.
```
Comment (ebattat) [inline docs/source/index.md:14]: change all to instance_run

**Review by ebattat — COMMENTED** [review]
Did you check that data distribute properly into Elastic/Grafana ?

**Comment by athiruma** [comment]
> Did you check that data distribute properly into Elastic/Grafana ?

Yes, see the grafana snap in the description.

File: cloud_governance/common/helpers/azure/azure_cleanup_operations.py (hunk 4 above)
Comment (ebattat) [reply cloud_governance/common/helpers/azure/azure_cleanup_operations.py]: can you change it in the diagram ?

**Comment by athiruma** [comment]
Policy Implementation-Policy Implementation (1)
Updated the diagram

**Comment by athiruma** [comment]
Old versions: [ Keeping here for the version tracking. ] 
Policy Implementation (1)
Policy Implementation-Policy Implementation

**Pull request merged**